pysat/utils/coords.py

```diff
@@ -103,12 +107,18 @@ def calc_solar_local_time(inst, lon_name=None, slt_name='slt'):
 
     # Convert from numpy epoch nanoseconds to UT seconds of day
     ut_hr = list()
+    first_time = inst.index.values[:1].astype(int)[0]
     for nptime in inst.index.values.astype(int):
+        # Account for any difference in days across loaded data (multi-day span)
+        day_diff = np.floor((nptime - first_time) * 1.0e-9 / 86400.)
+
```

Comment:
Wouldn't it be quicker to only do this if needed? You could just set it to zero if not needed.